expression/builtin_math.go

```diff
@@ -1163,11 +1164,25 @@ func (b *builtinConvSig) Clone() builtinFunc {
 // evalString evals CONV(N,from_base,to_base).
 // See https://dev.mysql.com/doc/refman/5.7/en/mathematical-functions.html#function_conv.
 func (b *builtinConvSig) evalString(row chunk.Row) (res string, isNull bool, err error) {
-	str, isNull, err := b.args[0].EvalString(b.ctx, row)
-	if isNull || err != nil {
-		return res, isNull, err
+	var str string
+	switch x := b.args[0].(type) {
+	case *Constant:
+		if x.Value.Kind() == types.KindBinaryLiteral {
+			str = x.Value.GetBinaryLiteral().ToBitLiteralString(true)
+		}
+	case *ScalarFunction:
+		if x.FuncName.L == ast.Cast {
+			arg0 := x.GetArgs()[0]
+			if arg0.GetType().Hybrid() || IsBinaryLiteral(arg0) {
+				str, isNull, err = arg0.EvalString(b.ctx, row)
+				if isNull || err != nil {
+					return str, isNull, err
+				}
+				d := types.NewStringDatum(str)
+				str = d.GetBinaryLiteral().ToBitLiteralString(true)
+			}
+		}
```

Comment:
what if the func is not Cast but produces string that needs to be converted?

Reply:
When we build `conv` function. If the first argument is NOT string type. The `cast` would be wrapped. No other cases. You can refer to this code.

```go
func (c *convFunctionClass) getFunction(ctx sessionctx.Context, args []Expression) (builtinFunc, error) {
	if err := c.verifyArgs(args); err != nil {
		return nil, err
	}

	bf, err := newBaseBuiltinFuncWithTp(ctx, c.funcName, args, types.ETString, types.ETString, types.ETInt, types.ETInt)
	if err != nil {
		return nil, err
	}
	bf.tp.Charset, bf.tp.Collate = ctx.GetSessionVars().GetCharsetInfo()
	bf.tp.Flen = 64
	sig := &builtinConvSig{bf}
	sig.setPbCode(tipb.ScalarFuncSig_Conv)
	return sig, nil
}
```